api/models/Centres.js

```diff
@@ -90,18 +90,24 @@ const model = {
       return response;
     }
   },
+
   getGenderAndCentreByCIDLocation: function (location) {
-    return this.findOne(
-      {
-        or: [
-          {male_cid_name: location},
-          {female_cid_name: location}
-        ]
-      })
-      .then(centre => centre ? {
-        centre: centre.id,
-        gender: centre.male_cid_name === location ? "male" : "female"
-      } : centre);
+    return this.find().then(centres =>
+      _.compact(_.map(centres, centre => {
```

Comment:
I did/do understand what you mean re: our conversation. And perhaps I can simply cache the call to `find`.